src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
 
-  public final SwerveDriveKinematics kinematics = new SwerveDriveKinematics(MODULE_OFFSET);
+  private final MotorControllerGroup rightMotors = new MotorControllerGroup(rightSparks);
+  private final MotorControllerGroup leftMotors = new MotorControllerGroup(leftSparks);
 
-  // Odometry and pose estimation
-  private final SwerveDrivePoseEstimator odometry;
+  // Gyros
+  @Log private final WPI_PigeonIMU gyro = new WPI_PigeonIMU(DrivePorts.GYRO_PORT);
+  //Right and Left side encoders
+  @Log private final RelativeEncoder rightEncoder = FRmotor.getEncoder();
+  @Log private final RelativeEncoder leftEncoder = FLmotor.getEncoder();
 
-  @Log private final Field2d field2d = new Field2d();
-  private final FieldObject2d[] modules2d;
+  @Log private double heading;
+  
+  @Log private final PIDController RdrivePID = new PIDController(DriveConstants.kP, DriveConstants.kI, DriveConstants.kD);
+  @Log private final PIDController LdrivePID = new PIDController(DriveConstants.kP, DriveConstants.kI, DriveConstants.kD);
 
-  // Rate limiting
-  private final SlewRateLimiter xLimiter = new SlewRateLimiter(MAX_ACCEL);
-  private final SlewRateLimiter yLimiter = new SlewRateLimiter(MAX_ACCEL);
+  @Log private final SimpleMotorFeedforward RdriveFF = new SimpleMotorFeedforward(DriveConstants.kS, DriveConstants.kV, DriveConstants.kA);
+  @Log private final SimpleMotorFeedforward LdriveFF = new SimpleMotorFeedforward(DriveConstants.kS, DriveConstants.kV, DriveConstants.kA);
 
-  @Log private double speedMultiplier = 1;
+  
 
-  public static Drive create() {
-    return Robot.isReal()
-        ? new Drive(
-            new MAXSwerveModule(
-                MODULE_NAMES[0], FRONT_LEFT_DRIVE, FRONT_LEFT_TURNING, ANGULAR_OFFSETS[0]),
-            new MAXSwerveModule(
-                MODULE_NAMES[1], FRONT_RIGHT_DRIVE, FRONT_RIGHT_TURNING, ANGULAR_OFFSETS[1]),
-            new MAXSwerveModule(
-                MODULE_NAMES[2], REAR_LEFT_DRIVE, REAR_LEFT_TURNING, ANGULAR_OFFSETS[2]),
-            new MAXSwerveModule(
-                MODULE_NAMES[3], REAR_RIGHT_DRIVE, REAR_RIGHT_TURNING, ANGULAR_OFFSETS[3]))
-        : new Drive(new SimModule(), new SimModule(), new SimModule(), new SimModule());
-  }
+  /**
+   * Encoders PID and FF controllers, -> mostly done by now, maybe there is something else about this that can be worked on later
+   * DifferentialDriveOdometry -> used DifferentialDrivePoseEstimator instead rn
+   * Getters for pose, encoder values
+   * Setters for setpoints and other important values
+   * Style
+   */
+  
 
-  public Drive(ModuleIO frontLeft, ModuleIO frontRight, ModuleIO rearLeft, ModuleIO rearRight) {
-    this.frontLeft = frontLeft;
-    this.frontRight = frontRight;
-    this.rearLeft = rearLeft;
-    this.rearRight = rearRight;
+  //sets up Odometry
+  private final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(ROBOT_TRACK);
+  DifferentialDriveWheelSpeeds speeds = new DifferentialDriveWheelSpeeds(0, 0);
```

Comment:
Not sure why you initialized this here (see comment I left in your periodic method)